src/screens/auth/VerifyEmail/VerifyEmail.tsx

```diff
@@ -1,30 +1,40 @@
 import { NavigationProp, useNavigation } from '@react-navigation/native';
-import React, { useEffect } from 'react';
+import React, { useCallback, useEffect } from 'react';
 import { View } from 'react-native';
 import { Text } from 'src/components';
 import { ScreenTemplate } from 'src/components/templates';
-import { RootStackParamList } from 'src/navigation';
+import { Props, RootStackParamList } from 'src/navigation';
+import { useAppDispatch } from 'src/store';
+import { AsyncThunks } from 'src/store/thunks';
 
 import styles from './VerifyEmail.style';
 
-const VerifyEmail = () => {
+const VerifyEmail = ({ route }: Props) => {
   const navigation = useNavigation<NavigationProp<RootStackParamList>>();
+  const dispatch = useAppDispatch();
+  const [optionalMessage, setOptionalMessage] = React.useState('Waiting...');
+
+  const data = route.params?.data;
+  const verification = useCallback(async () => {
+    const response = await dispatch(AsyncThunks.verifyEmail(data));
+    if (response.payload?.status === 200) {
+      setOptionalMessage('Link sent to your email adress \nYou can sign in after verification');
+      navigation.navigate('Signin');
+    } else {
+      setOptionalMessage(response.payload?.data.message);
+      navigation.navigate('Main');
+    }
+  }, [optionalMessage]);
```

Comment:
I have added eslint rule which is checking the rules of hooks. Fix deps.

Check this part of the documentation about the parameters of `useEffect` and `useCallback` hooks.
https://react.dev/reference/react/useEffect#parameters
https://react.dev/reference/react/useCallback#parameters